bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/JellyfinHandlerFactory.java

```diff
@@ -0,0 +1,143 @@
+
+    @Activate
+    public JellyfinHandlerFactory(@Reference HttpService httpService) {
+        this.httpService = httpService;
+    }
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+        if (THING_TYPE_SERVER.equals(thingTypeUID)) {
+            var serverHandler = new JellyfinServerHandler((Bridge) thing);
+            registerAuthenticationServlet(serverHandler);
+            registerDiscoveryService(serverHandler);
+            return serverHandler;
+        }
+        if (THING_TYPE_CLIENT.equals(thingTypeUID)) {
+            return new JellyfinClientHandler(thing);
+        }
+        return null;
+    }
+
+    @Override
+    protected synchronized void removeHandler(ThingHandler thingHandler) {
+        if (thingHandler instanceof JellyfinServerHandler) {
+            var serverHandler = (JellyfinServerHandler) thingHandler;
+            unregisterAuthenticationServlet(serverHandler);
+            unregisterDiscoveryService(serverHandler);
+        }
+        super.removeHandler(thingHandler);
+    }
+
+    private synchronized void registerAuthenticationServlet(JellyfinServerHandler bridgeHandler) {
+        var auth = new JellyfinBridgeServlet(bridgeHandler);
+        try {
+            httpService.registerServlet(getAuthenticationServletPath(bridgeHandler), auth, null,
+                    httpService.createDefaultHttpContext());
+        } catch (NamespaceException | ServletException e) {
+            logger.warn("Register servlet fails", e);
+        }
+        servletRegistrations.put(bridgeHandler.getThing().getUID(), auth);
+    }
+
+    private synchronized void unregisterAuthenticationServlet(JellyfinServerHandler bridgeHandler) {
+        var loginServlet = servletRegistrations.get(bridgeHandler.getThing().getUID());
+        if (loginServlet != null) {
+            httpService.unregister(getAuthenticationServletPath(bridgeHandler));
+        }
+    }
+
+    private String getAuthenticationServletPath(JellyfinServerHandler bridgeHandler) {
+        return new StringBuilder().append("/").append(BINDING_ID).append("/")
+                .append(bridgeHandler.getThing().getUID().getId()).toString();
+    }
+
+    private synchronized void registerDiscoveryService(JellyfinServerHandler bridgeHandler) {
+        JellyfinClientDiscoveryService devicesDiscovery = new JellyfinClientDiscoveryService(bridgeHandler);
+        devicesDiscovery.activate();
+        var registration = bundleContext.registerService(DiscoveryService.class.getName(), devicesDiscovery,
+                new Hashtable<>());
+        discoveryServiceRegistrations.put(bridgeHandler.getThing().getUID(), registration);
+    }
```

Comment:
Did you take a look at the `ThingHandlerService`? That would make the code much simpler. See https://www.openhab.org/docs/developer/bindings/#discovery-that-is-bound-to-a-bridge